source/extensions/filters/http/fault/fault_filter.cc

```diff
@@ -375,17 +402,18 @@ void FaultFilter::postDelayInjection(const Http::RequestHeaderMap& request_heade
   // Delays can be followed by aborts
   const auto abort_code = abortHttpStatus(request_headers);
   if (abort_code.has_value()) {
-    abortWithHTTPStatus(abort_code.value());
+    abortWithStatus(abort_code.value(), abortGrpcStatus(request_headers));
```

Comment:
we currently call `abortGrpcStatus` twice every time grpc fault injection is enabled - any way we could limit the number of these calls to 1?